url7895f5cfa7_edm92_defaultlogic_tgz-pJ8-be_fnord_DefaultLogic_AppJ8/src/be/fnord/DefaultLogic/ExtraExample.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * A special example for a special someone
+ * 
+ * Cosmological considerations of penguins
+ * 
+ * 
+ * 
+ * 
+ * 
+Given the world: 
+	tweety_is_a_bird
+ And the rules 
+	[(tweety_is_a_bird):(tweety_can_fly) ==> (tweety_can_fly)]
+Possible Extensions
+	 Ext: Th(W U (tweety_can_fly))
+	= tweety_can_fly & tweety_is_a_bird 
+We one day learn that tweety is penguin.
+Given the world: 
+	tweety_is_a_bird & tweety_is_a_penguin & (tweety_is_a_penguin -> ~tweety_can_fly)
+ And the rules 
+	[(tweety_is_a_bird):(tweety_can_fly) ==> (tweety_can_fly)] , [(tweety_is_a_bird):(tweety_is_a_penguin) ==> (~tweety_can_fly)]
+Possible Extensions
+	 Ext: Th(W U (~tweety_can_fly))
+	= ~tweety_can_fly & tweety_is_a_bird & tweety_is_a_penguin & (~tweety_can_fly | ~tweety_is_a_penguin) 
+ */
+
+package be.fnord.DefaultLogic;
+
+import java.util.HashSet;
+
+import be.fnord.util.logic.DefaultReasoner;
+import be.fnord.util.logic.WFF;
+import be.fnord.util.logic.defaultLogic.DefaultRule;
+import be.fnord.util.logic.defaultLogic.RuleSet;
+import be.fnord.util.logic.defaultLogic.WorldSet;
+
+public class ExtraExample {
+
+	// Example 2.9 From
+	// J. P. Delgrande, T. Schaub, and W. K. Jackson (1994). Alternative
+	// approaches to default logic. Artificial Intelligence, 70:167-237.
+
+	public static void example() {
+		WorldSet myWorld = new WorldSet();
+		// Start by believing that tweety is a bird
+		myWorld.addFormula("tweety_is_a_bird"); // We think that x is a bird
+		
+
+		DefaultRule rule1 = new DefaultRule();
+		rule1.setPrerequisite("tweety_is_a_bird");
+		rule1.setJustificatoin("tweety_can_fly");
+		rule1.setConsequence("tweety_can_fly");
+		
+		RuleSet myRules = new RuleSet();
+		myRules.addRule(rule1);
+//		myRules.addRule(rule2);
+
+		DefaultReasoner loader = new DefaultReasoner(myWorld, myRules);
+		HashSet<String> extensions = loader.getPossibleScenarios();
+
+		a.e.println("Given the world: \n\t" + myWorld.toString()
+				+ "\n And the rules \n\t" + myRules.toString());
+
+		a.e.println("Possible Extensions");
+		for (String c : extensions) {
+			a.e.println("\t Ext: Th(W U (" + c + "))");
+			// Added closure operator
+			a.e.incIndent();
+			WFF world_and_ext = new WFF("(( " + myWorld.getWorld() + " ) & ("
+					+ c + "))");
+			a.e.println("= " + world_and_ext.getClosure());
+			a.e.decIndent();
+
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 65:74, 112:121.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.